test/test_serialization.py

```diff
@@ -208,3 +219,170 @@ def echo(self, arg):
     msg = get_messages()[0]
     assert '"RESULT": "HELLO"' in msg['payload']
     assert msg['properties']['content_type'] == "application/x-upper-json"
+
+
+def test_consumer_accepts_multiple_serialization_formats(
+    container_factory, rabbit_config, rabbit_manager
+):
+
+    consumed = Mock()
+
+    class Service(object):
+
+        name = 'service'
+
+        @consume(queue=Queue(exchange=Exchange('spam'), name='some-queue'))
+        def consume(self, payload):
+            assert payload == {'spam': 'ham'}
+            consumed(payload)
+
+    rabbit_config[ACCEPT_CONFIG_KEY] = ['json', 'yaml']
+
+    container = container_factory(Service, rabbit_config)
+    container.start()
+
+    payload = {'spam': 'ham'}
+
+    def publish(serialized_payload, content_type):
+        rabbit_manager.publish(
+            rabbit_config['vhost'], 'spam', '', serialized_payload,
+            properties={'content_type': content_type})
+
+    publish(json.dumps(payload), 'application/json')
+    publish(yaml.dump(payload), 'application/x-yaml')
+
+    assert consumed.mock_calls == [call(payload), call(payload)]
+
+
+def test_rpc_accepts_multiple_serialization_formats(
+    container_factory, rabbit_config, rabbit_manager,
+    sniffer_queue_factory
+):
+
+    called = Mock()
+
+    class ForwardingService(object):
+
+        name = 'forwarder'
+
+        echoer = RpcProxy('echoer')
+
+        @rpc
+        def forward(self, payload):
+            return self.echoer.echo(payload)
+
+    class EchoingService(object):
+
+        name = 'echoer'
+
+        @rpc
+        def echo(self, payload):
+            called(payload)
+            return payload
+
+    # Echoer serialiser is set to JSON, but accepts both JSON and YAML
+    echoer_config = rabbit_config.copy()
+    echoer_config[SERIALIZER_CONFIG_KEY] = 'json'
+    echoer_config[ACCEPT_CONFIG_KEY] = ['json', 'yaml']
+
+    echoer = container_factory(EchoingService, echoer_config)
+    echoer.start()
+
+    payload = {'spam': 'ham'}
+
+    # Forwarder serialiser is set to JSON and should send and receive JSON
```

Comment:
Might be worth splitting this into multiple tests. If you put them all into a class you can do the setup in fixtures to avoid duplication

Reply:
Sure thing, parameterised.